instrumentation/rocketmq/rocketmq-client/rocketmq-client-5.0/testing/src/main/java/io/opentelemetry/instrumentation/rocketmqclient/v5_0/AbstractRocketMqClientTest.java

```diff
@@ -38,57 +52,79 @@ public abstract class AbstractRocketMqClientTest {
             .withEnv("rocketmq.namesrv.port", String.valueOf(namesrvPort))
-            .withExposedPorts(proxyPort)) {
-      // Start the container.
-      container.start();
-      String endpoints = "127.0.0.1:" + proxyPort;
-      ClientConfiguration clientConfiguration =
-          ClientConfiguration.newBuilder().setEndpoints(endpoints).build();
-      // Inner topic of the container.
-      String topic = "normal-topic-0";
-      ClientServiceProvider provider = ClientServiceProvider.loadService();
-      Producer producer =
+            .withExposedPorts(proxyPort);
+    // Start the container.
+    container.start();
+  }
+
+  @AfterAll
+  static void tearDown() {
+    container.close();
+  }
+
+  @Test
+  public void testSendAndConsumeMessage() throws ClientException, IOException {
+    ClientConfiguration clientConfiguration =
+        ClientConfiguration.newBuilder().setEndpoints(endpoints).build();
+    // Inner topic of the container.
+    String topic = "normal-topic-0";
+    ClientServiceProvider provider = ClientServiceProvider.loadService();
+    String consumerGroup = "group-normal-topic-0";
+    String tag = "tagA";
+    FilterExpression filterExpression = new FilterExpression(tag, FilterExpressionType.TAG);
+    try (PushConsumer ignored =
+        provider
+            .newPushConsumerBuilder()
+            .setClientConfiguration(clientConfiguration)
+            .setConsumerGroup(consumerGroup)
+            .setSubscriptionExpressions(Collections.singletonMap(topic, filterExpression))
+            .setMessageListener(messageView -> ConsumeResult.SUCCESS)
+            .build()) {
+      try (Producer producer =
           provider
               .newProducerBuilder()
               .setClientConfiguration(clientConfiguration)
               .setTopics(topic)
-              .build();
+              .build()) {
 
-      String tag = "tagA";
-      String[] keys = new String[] {"yourMessageKey-0", "yourMessageKey-1"};
-      byte[] body = "foobar".getBytes(StandardCharsets.UTF_8);
-      Message message =
-          provider
-              .newMessageBuilder()
-              .setTopic(topic)
-              .setTag(tag)
-              .setKeys(keys)
-              .setBody(body)
-              .build();
+        String[] keys = new String[] {"yourMessageKey-0", "yourMessageKey-1"};
+        byte[] body = "foobar".getBytes(StandardCharsets.UTF_8);
+        Message message =
+            provider
+                .newMessageBuilder()
+                .setTopic(topic)
+                .setTag(tag)
+                .setKeys(keys)
+                .setBody(body)
+                .build();
 
-      SendReceipt sendReceipt = producer.send(message);
-      testing()
-          .waitAndAssertTraces(
-              traceAssert ->
-                  traceAssert.hasSpansSatisfyingExactly(
-                      spanDataAssert ->
-                          spanDataAssert
+        SendReceipt sendReceipt = producer.send(message);
```

Comment:
Similar to the consumer comment: we usually call the producer with some parent context to make sure that it's being properly propagated.
```suggestion
        SendReceipt sendReceipt = testing.runWithSpan("producer", () -> producer.send(message));
```